### ISSUE TITLE
Configure Sections using markdown files in folders and front-matter

### ISSUE THREAD

**Issue by pke** [styleguidist/react-styleguidist#365]
I know we can configure sections via a custom config.

But wouldn't it be cool if the sections could be generated simply by `[index|section].md` files in src folders using YAML fron matter (https://github.com/matthewwithanm/markdown-with-front-matter-loader)?

```
---
section: Name of the section
---

# Section description
```

This would further simplify the configuration. What you think?

**Comment by n1313** [comment]
I personally would not use the word "simply" to describe a setup with non-standard markdown syntax and an extra dependency which does exactly the same as one line in the config, but that's just me.

**Comment by okonet** [comment]
Front matter is just a convention that many are using. I'm not sure how much value it will bring in this context, though. @pke can you give us some use cases for that?

**Comment by sapegin** [comment]
Yeah, I’d like to see some use cases where it actually simplify configuration. For me it brings more questions: ordering for example?

What we can do is ignore front matters in markdown files and then generating sections config would be a few lines of code and up to the user.

**Comment by n1313** [comment]
While this extra markup could definitely be ignored by RSG, it will remain in README.md files and "pollute" their "native" display in GitHub/Stash. I personally quite enjoy having my code and my examples being written in a simple, no-special-markup-required way. But, I guess, if this is an optional feature then I don't have to use it if I don't want to, so that's not a big deal.

**Comment by okonet** [comment]
GitHub actually can present front matter very nicely as a table. See https://github.com/okonet/colorsnapper-website/blob/master/_pages/faq.md for example so this isn't an issue IMO.

**Comment by sapegin** [comment]
It can be optional in a way that you don’t have to use front matters. So Styleguidist will just ignore them but actual reading front matters and generation of a config object would be up to the user.

**Comment by PDXIII** [comment]
I would love to be able to use front matter.

**Comment by okonet** [comment]
Yeah the more I think about it the more I like the idea. I might work on a PR for that since I might need it for some of my projects

**Comment by okonet** [comment]
My use case is a customization of particular examples. So what I’d be looking for is an ability to pass props to the example renderer component or to the example components themselves (via context?)

**Comment by PDXIII** [comment]
I would like to have it to be able to give components an approval status, which could be passed to the header. Or add an author etc.

**Comment by okonet** [comment]
That sounds great but I’d suggest to use frontmatter instead of HTML comments. Possible?

**Comment by MrHavard** [comment]
> That sounds great but I’d suggest to use frontmatter instead of HTML comments. Possible?

We're using gray-matter which uses frontmatter but also allows us to define custom delimiters, which we specified as html comments that wrap the yaml. We were going to use frontmatter directly but having a table pop up in the top of our readme's in github wasn't something we liked.

Example
```
<!---
title: Accordion Group
parent: Accordion
--->
```
